src/Internal/Plutip/Server.purs

```diff
@@ -124,24 +139,97 @@ withPlutipContractEnv
   -> (ContractEnv () -> wallets -> Aff a)
   -> Aff a
 withPlutipContractEnv plutipCfg distr cont = do
+  cleanupRef <- liftEffect $ Ref.new mempty
+  Aff.bracket
+    (startPlutipContractEnv plutipCfg distr cleanupRef)
+    (const $ cleanup cleanupRef)
+    \{ env, wallets, printLogs } -> do
+      runWithLogs printLogs (cont env wallets)
+
+-- | Run `Contract`s in tests in a single Plutip instance.
+testPlutipContracts
+  :: forall (distr :: Type) (wallets :: Type)
+   . UtxoDistribution distr wallets
+  => PlutipConfig
+  -> distr
+  -> TestPlanM (wallets -> Contract () Unit) Unit
+  -> TestPlanM (Aff Unit) Unit
+testPlutipContracts plutipCfg distr tests = do
+  let groupedTests = group "HIDDEN_GROUP" tests
```

Comment:
What's the point of this groupedTests thing?

Reply:
Unfortunately Mote's bracket semantics is per every group or test in the current 'layer': https://pursuit.purescript.org/packages/purescript-mote/3.0.0/docs/Mote#v:bracket

Imo it's a bit of an odd choice on their end, so this ensures we only run the bracket once

Reply:
Please add a long comment about that

Reply:
I've decided to drop the group and explain the caveats in a doc comment. We don't export our `planToSpec`, so the special casing would be odd for outside users of `testPlutipContracts`